src/api/middleware/company.js

```diff
@@ -94,7 +94,32 @@ const isNotBlocked = (owner) => async (req, res, next) => {
             ValidationReasons.COMPANY_BLOCKED
         ));
     }
+    return next();
+};
 
+const isNotDisabled = (owner) => async (req, res, next) => {
+    const company = await (new CompanyService()).findById(owner, true);
+    if (company.isDisabled) {
+        return next(new APIError(
+            HTTPStatus.FORBIDDEN,
+            ErrorTypes.FORBIDDEN,
+            ValidationReasons.COMPANY_DISABLED
+        ));
+    }
+    return next();
+};
+
+const validCompany = (companyId) => async (req, res, next) => {
```

Comment:
It seems like this function name does not clearly describe what it does (in reality it checks for permissions). Isn't there any function that does this check already? Maybe cc @DoStini or @BrunoRosendo (?)

Reply:
Hmm, I think we only have a function that checks permissions for owning a offer or simply checking if the user is a company, we never had to verify if they have perms for modifying a company (themselves). The closest feature was the block/unblock but that's only for admins/gods.

Maybe we could move this logic to the auth middleware with another name and call it in the endpoint's [auth checks](https://github.com/NIAEFEUP/nijobs-be/blob/74fad3988e597d6980ff09403facdf02744e8c10/src/api/routes/company.js#L110)?

Reply:
https://github.com/NIAEFEUP/nijobs-be/blob/6ad5ea03cfe2a2c6fdf1f487ed48c22c7ac04581/src/api/middleware/validators/offer.js#L511

This is what I was recalling. It's not quite the same thing, as it has logic specific to the hiddenReason and does not check ownership. In this case, there's no need to generalize I think, but I still would like for this to have a better name